Explorer/Assets/DCL/SDKComponents/AvatarShape/Systems/UpdateAvatarShapeTargetPositionSystem.cs

```diff
@@ -0,0 +1,80 @@
+using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using CrdtEcsBridge.Components.Transform;
+using DCL.Character.Components;
+using DCL.Diagnostics;
+using DCL.ECSComponents;
+using DCL.SDKComponents.Tween.Components;
+using ECS.Abstract;
+using ECS.Groups;
+using ECS.Unity.AvatarShape.Components;
+using ECS.Unity.Transforms.Systems;
+
+namespace DCL.SDKComponents.AvatarShape.Systems
+{
+    [UpdateInGroup(typeof(SyncedSimulationSystemGroup))]
+    [UpdateBefore(typeof(UpdateTransformSystem))]
+    [LogCategory(ReportCategory.AVATAR)]
+    public partial class UpdateAvatarShapeTargetPositionSystem : BaseUnityLoopSystem
+    {
+        private readonly World globalWorld;
+
+        private UpdateAvatarShapeTargetPositionSystem(World world, World globalWorld) : base(world)
+        {
+            this.globalWorld = globalWorld;
+        }
+
+        protected override void Update(float t)
+        {
+            UpdateAvatarTargetPositionQuery(World);
+            UpdateAvatarWithTweenTargetPositionQuery(World);
+        }
+
+        [Query]
+        [None(typeof(SDKTweenComponent))]
+        private void UpdateAvatarTargetPosition(ref SDKAvatarShapeComponent sdkAvatarShapeComponent, ref SDKTransform sdkTransform)
```

Comment:
Let's change all the `ref` for components that you are not going to modify and just use `in`